mm2src/coins/z_coin/storage.rs

```diff
@@ -1,5 +1,201 @@
+use crate::z_coin::{ValidateBlocksError, ZcoinConsensusParams};
+
 pub mod blockdb;
 pub use blockdb::*;
 
 pub mod walletdb;
 pub use walletdb::*;
+
+#[cfg(target_arch = "wasm32")]
+use walletdb::wasm::storage::DataConnStmtCacheWasm;
+#[cfg(debug_assertions)]
+use zcash_client_backend::data_api::error::Error;
+use zcash_client_backend::data_api::PrunedBlock;
+use zcash_client_backend::proto::compact_formats::CompactBlock;
+use zcash_client_backend::wallet::{AccountId, WalletTx};
+use zcash_client_backend::welding_rig::scan_block;
+#[cfg(not(target_arch = "wasm32"))]
+use zcash_client_sqlite::for_async::DataConnStmtCacheAsync;
+use zcash_extras::{WalletRead, WalletWrite};
+use zcash_primitives::block::BlockHash;
+use zcash_primitives::consensus::BlockHeight;
+use zcash_primitives::merkle_tree::CommitmentTree;
+use zcash_primitives::sapling::Nullifier;
+use zcash_primitives::zip32::ExtendedFullViewingKey;
+
+#[derive(Clone)]
+pub struct DataConnStmtCacheWrapper {
+    #[cfg(not(target_arch = "wasm32"))]
+    pub cache: DataConnStmtCacheAsync<ZcoinConsensusParams>,
+    #[cfg(target_arch = "wasm32")]
+    pub cache: DataConnStmtCacheWasm,
+}
+
+impl DataConnStmtCacheWrapper {
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn new(cache: DataConnStmtCacheAsync<ZcoinConsensusParams>) -> Self { Self { cache } }
+    #[cfg(target_arch = "wasm32")]
+    pub fn new(cache: DataConnStmtCacheWasm) -> Self { Self { cache } }
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn inner(&self) -> DataConnStmtCacheAsync<ZcoinConsensusParams> { self.clone().cache }
+    #[cfg(target_arch = "wasm32")]
+    pub fn inner(&self) -> DataConnStmtCacheWasm { self.clone().cache }
```

Comment:
The inner function pattern should return a reference. The functions here should be inlined as well.